src/twisted/conch/ssh/filetransfer.py

```diff
@@ -527,6 +531,21 @@ def connectionMade(self):
         self.sendPacket(FXP_INIT, data)
 
 
+    def connectionLost(self, reason):
+        """
+        Abort any pending requests.
+        """
+
+        # On Python2, we're an oldstyle class, so we can't use super().
+        FileTransferBase.connectionLost(self, reason)
+
+        # If there are still requests waiting for responses when the
+        # connection is lost, fail them.
+        while self.openRequests:
+            _, deferred = self.openRequests.popitem()
+            deferred.errback(reason)
```

Comment:
I don't get a different exception, but in my opinion that's a bug in Conch — it should provide a different exception if the connection fails in a different way. So I wanted to pass through the exception in case it someday becomes meaningful.

Perhaps this code should pass through the exception, *unless* it is a `ConnectionDone`, in which case it should be converted to `ConnectionLost` (since it cannot be a clean closure if there are unfinished requests)?

Reply:
Well... I tracked this and from what I can see this is a low error in Twisted client... I am seeing similar errors in Twisted Web client.

But I am OK to accept this... I would still like to see a different exeption here, as I imagine that many other people using this will have to patch this code in their own code.

Reply:
I updated the PR to convert a ConnectionDone into a ConnectionLost.